app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/CreateDLCOfferDialog.scala

```diff
@@ -35,12 +43,56 @@ class CreateDLCOfferDialog extends Logging {
     dialog.dialogPane().stylesheets = GlobalData.currentStyleSheets
     dialog.resizable = true
 
-    val fields: mutable.Map[Int, (TextField, TextField)] = mutable.Map.empty
+    val vbox = buildView(hex, null, None)
 
-    var announcementDetailsShown = false
-    val announcementOrContractInfoTF = new TextField()
-    var decompOpt: Option[DigitDecompositionEventDescriptorV0TLV] = None
+    dialog.dialogPane().content = new ScrollPane {
+      content = vbox
+    }
 
+    // When the OK button is clicked, convert the result to a CreateDLCOffer.
+    dialog.resultConverter = dialogButton =>
+      if (dialogButton == ButtonType.OK) {
+        getCliCommand()
+      } else None
+
+    val result = dialog.showAndWait()
+
+    result match {
+      case Some(Some(offer: CreateDLCOffer)) => Some(offer)
+      case Some(_) | None                    => None
+    }
+  }
+
+  val fields: mutable.Map[Int, (TextField, TextField)] = mutable.Map.empty
+
+  var announcementDetailsShown = false
+  lazy val announcementOrContractInfoTF = new TextField()
+  var decompOpt: Option[DigitDecompositionEventDescriptorV0TLV] = None
+
+  val pointMap: scala.collection.mutable.Map[
+    Int,
+    (TextField, TextField, CheckBox)] =
+    scala.collection.mutable.Map.empty
+
+  val roundingMap: scala.collection.mutable.Map[Int, (TextField, TextField)] =
+    scala.collection.mutable.Map.empty
+
+  lazy val feeRateTF = new TextField() {
+    text = GlobalData.feeRate.toLong.toString
+    promptText = "(optional)"
+  }
+
+  lazy val collateralTF = new TextField() {
+    promptText = "Satoshis"
+  }
+  setNumericInput(collateralTF)
+
+  lazy val refundDatePicker = new DatePicker()
+
+  def buildView(
+      initialContract: String = "",
```

Comment:
the convention is to make the parameter that has a default argument last, i.e. 

```scala
def buildView(
  announcement: OracleAnnouncementV0TLV,
  contractInfoOpt: Option[ContractInfoV0TLV],
  initialContract: String = ""
)
```


also with methods please make sure you annotate the return type.